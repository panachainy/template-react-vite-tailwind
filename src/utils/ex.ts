export function add(input1: number, input2: number): number {
  return input1 + input2
}
