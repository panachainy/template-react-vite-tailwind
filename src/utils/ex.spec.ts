import { describe, it, expect } from 'vitest'
import { add } from './ex'

describe('add', () => {
  it('should match the snapshot for various inputs', () => {
    const results = [
      { input: [2, 3], output: add(2, 3) },
      { input: [5, -3], output: add(5, -3) },
      { input: [-4, -6], output: add(-4, -6) },
      { input: [0, 0], output: add(0, 0) },
      { input: [0, 7], output: add(0, 7) },
      { input: [9, 0], output: add(9, 0) },
      { input: [100, 200], output: add(100, 200) },
      { input: [-50, 50], output: add(-50, 50) },
    ]
    expect(results).toMatchSnapshot()
  })
})
