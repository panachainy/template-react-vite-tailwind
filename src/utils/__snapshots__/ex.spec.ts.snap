// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`add > should match the snapshot for various inputs 1`] = `
[
  {
    "input": [
      2,
      3,
    ],
    "output": 5,
  },
  {
    "input": [
      5,
      -3,
    ],
    "output": 2,
  },
  {
    "input": [
      -4,
      -6,
    ],
    "output": -10,
  },
  {
    "input": [
      0,
      0,
    ],
    "output": 0,
  },
  {
    "input": [
      0,
      7,
    ],
    "output": 7,
  },
  {
    "input": [
      9,
      0,
    ],
    "output": 9,
  },
  {
    "input": [
      100,
      200,
    ],
    "output": 300,
  },
  {
    "input": [
      -50,
      50,
    ],
    "output": 0,
  },
]
`;
